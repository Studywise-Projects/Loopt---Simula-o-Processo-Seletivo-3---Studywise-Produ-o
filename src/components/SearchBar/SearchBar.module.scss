@import '@/styles/_global.scss';

.stack {
  align-items: flex-end;
  display: flex;
  flex-direction: row;
  gap: 16px;
  height: 44px;
  width: 320px;
  
  @media only screen and (min-width: 768px) {
    width: 420px;
  }
}

.auxSearch {
  flex: 1;
  width: auto;
}

.input {
  width: 90%;
}

.iconButton {
  padding: 0;
}

.icon {
  color: $white;
  border-radius: 100%;
  background-color: $primary-color;
  padding: 4px;
}