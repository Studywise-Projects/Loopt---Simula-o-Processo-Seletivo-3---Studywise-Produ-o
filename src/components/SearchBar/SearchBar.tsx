import SearchIcon from '@mui/icons-material/Search';
import { IconButton, Stack, TextField } from '@mui/material';
import { ChangeEventHandler } from 'react';
import styles from './SearchBar.module.scss';

interface ISearchBar {
  id: string;
  label: string;
  value: string;
  handleChange: ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
}

function SearchBar({ id, label, value, handleChange }: ISearchBar) {
  return (
    <Stack className={styles.stack}>
      <TextField
        data-testId='searchbar'
        variant='standard'
        id={id}
        type='search'
        label={label}
        value={value}
        onChange={handleChange}
        className={styles.input}
      />
      <IconButton className={styles.iconButton}>
        <SearchIcon sx={{ fontSize: '24px' }} className={styles.icon} />
      </IconButton>
    </Stack>
  );
}

export default SearchBar;
