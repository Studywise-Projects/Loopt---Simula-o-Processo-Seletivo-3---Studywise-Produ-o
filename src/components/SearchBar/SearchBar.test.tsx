import { fireEvent, render } from '@testing-library/react';
import SearchBar from './SearchBar';

describe('SearchBar', () => {
  it('Should match the snapshot', () => {
    const { getByLabelText } = render(
      <SearchBar
        id='search-bar-test'
        label='test'
        value=''
        handleChange={() => undefined}
      />,
    );

    const element = getByLabelText('test');
    expect(element).toMatchSnapshot();
  });

  it('Should render correctly', () => {
    const { getByLabelText } = render(
      <SearchBar
        id='search-bar-test'
        label='test'
        value=''
        handleChange={() => undefined}
      />,
    );

    const element = getByLabelText('test');
    expect(element).toBeInTheDocument();
  });

  it('Should call handleChangeMock function call', () => {
    const handleChangeMock = jest.fn();

    const { getByLabelText } = render(
      <SearchBar
        id='search-bar-test'
        label='test'
        value=''
        handleChange={handleChangeMock}
      />,
    );

    const element = getByLabelText('test');
    fireEvent.change(element, { target: { value: 'search bar changed' } });
    expect(handleChangeMock).toHaveBeenCalledTimes(1);
  });
});
