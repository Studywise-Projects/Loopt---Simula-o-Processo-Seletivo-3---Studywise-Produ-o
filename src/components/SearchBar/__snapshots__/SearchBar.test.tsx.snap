// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Should match the snapshot 1`] = `
<input
  aria-invalid="false"
  class="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch css-c63i49-MuiInputBase-input-MuiInput-input"
  id="search-bar-test"
  type="search"
  value=""
/>
`;
